src/sentry/tasks/relay.py

```diff
@@ -32,6 +32,23 @@ def update_config_cache(
     from sentry.relay import projectconfig_cache
     from sentry.relay.config import get_project_config
 
+    # At this point, we haven't marked the task as completed, so it should be
+    # debounced. If it isn't, another worker has picked a task for the same
+    # public key and marked it as done. Since we want to avoid recomputation,
+    # exit the current task.
+    if not projectconfig_debounce_cache.is_debounced(
+        public_key=public_key, project_id=None, organization_id=None
+    ):
+        return
+
+    # Delete key before generating configs such that we never have an outdated
+    # but valid cache.
+    #
+    # If this was running at the end of the task, it would be more effective
+    # against bursts of updates, but introduces a different race where an
+    # outdated cache may be used.
+    projectconfig_debounce_cache.mark_task_done(public_key, project_id, organization_id)
```

Comment:
Is there a specific reason to move this up? Previously, if `mark_task_done` fails for instance due to Redis issues, we would have organization and project information on the resulting Sentry error. Now, we no longer have this data. It would be better to keep the former order.